File: Leetcode/Arrays/4summII.454.py
from typing import List

class Solution:
    def fourSumCount(self, nums1: List[int], nums2: List[int], nums3: List[int], nums4: List[int]) -> int:
        result = 0
        dict = {}
        for n1 in nums1:
            for n2 in nums2:
                if n1+n2 in dict:
                    dict[n1+n2]= dict[n1+n2]+1
                else:
                    dict[n1+n2] = 1

        for n3 in nums3:
            for n4 in nums4:
                target = -(n3 + n4)
                if target in dict:
                    result += dict[target]
        return result

File: Leetcode/Arrays/4summII.454.go
package Arrays

//Given four integer arrays nums1, nums2, nums3, and nums4 all of length n, return the number of tuples (i, j, k, l) such that:
//
//0 <= i, j, k, l < n
//nums1[i] + nums2[j] + nums3[k] + nums4[l] == 0

func fourSumCount(nums1 []int, nums2 []int, nums3 []int, nums4 []int) int {
	var result int
	//m1 := make(map[int]int)
	var m1 map[int]int
	for _, n1 := range nums1 {
		for _, n2 := range nums2 {
			m1[n1+n2]++
		}
	}
	for _, n3 := range nums3 {
		for _, n4 := range nums4 {
			sum := n3+n4
			if val,ok := m1[-sum]; ok {
				result += val
			}

		}
	}
	return result
}
